--- satyr-0.14/lib/core_unwind_elfutils.c.orig	2014-03-26 16:23:25.000000000 +0100
+++ satyr-0.14/lib/core_unwind_elfutils.c	2014-06-14 16:40:38.064971360 +0200
@@ -166,7 +166,7 @@
             set_error_dwfl("dwfl_getthreads");
         else if (ret == DWARF_CB_ABORT)
         {
-            set_error(thread_arg.error_msg);
+            set_error("%s", thread_arg.error_msg);
             free(thread_arg.error_msg);
         }
         else
